.github/workflows/code-health.yml

```diff
@@ -16,22 +16,34 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - name: Set up Go
-      uses: actions/setup-go@v5
+    - uses: actions/setup-go@v5
       with:
         go-version-file: 'go.mod'
+    - name: Install tools for mocks and linter
+      run: make tools
+    - name: Mock generation
+      run: mockery
+    - name: Check for uncommited files
+      run: |
+        export FILES=$(git ls-files -o -m --directory --exclude-standard --no-empty-directory)
+        export LINES=$(echo "$FILES" | awk 'NF' | wc -l)
+        if [ $LINES -ne 0 ]; then
+          echo "Detected files that need to be committed:"
+          echo "$FILES" | sed -e "s/^/  /"
+          echo ""
+          echo "Mock skeletons are not up-to-date, you may have forgotten to run mockery before committing your changes."
+          exit 1
+        fi
     - name: Build
       run: make build
+    - name: Linter
+      run: make lint
   unit-test:
     needs: build
     runs-on: ubuntu-latest
-    permissions:
-      pull-requests: write
```

Comment:
why are we weakening the permissions?

Reply:
@gssbzn the intention was to simplify the action, what problems or risks do you see removing this?

Reply:
at some point we'll have to do some audit of gh actions permissions and be sure GH actions only use what they need, I'm not sure when was this permission added or why, a blame to understand the context would be great before trying to change it 

I started some of this auditing on the CLI using https://github.com/ossf/scorecard but it's quite consuming and I haven't had the time to dig deeper on it

Reply:
i see, thanks for the info, it looks like we'll need to a focused effort on this and change all workflows.